.github/workflows/snap.yaml

```diff
@@ -0,0 +1,113 @@
+name: snap
+run-name: Build and publish snap
+on:
+  pull_request:
+    branches:
+      - main
+      - 'release/**'
+  push:
```

Comment:
I think that using both `pull_request` and `push` trigger events will make the build/tests run twice, one once the PR is proposed and then when it gets merged, which probably we don't want, at least for main branch.